beamer/config/config.py

```diff
@@ -0,0 +1,55 @@
+from dataclasses import dataclass
+from beamer.typing import ChainId
+
+
+@dataclass(frozen=True)
+class BeamerConfig:
+    chains: dict[str, dict[str, ChainId | None]]
+    type2_chains: tuple[str, ...]
+    optimistic_chains: tuple[str, ...]
+
+    def _get_network_chain_ids(
+        self, chain_map: dict[str, ChainId | None], exclude_local_chains: bool
+    ) -> list[ChainId]:
+        chain_ids = []
+        for network, chain_id in chain_map.items():
+            if network == "local" and exclude_local_chains or chain_id is None:
+                pass
+            else:
+                assert chain_id is not None
+                chain_ids.append(chain_id)
+        return chain_ids
+
+    def filter_chain_ids_by_category_name(
+        self, category_name: str, exclude_local_chains: bool = True
+    ) -> list[ChainId]:
+        chain_ids = []
+        for chain in getattr(self, category_name, []):
+            chain_ids.extend(self._get_network_chain_ids(self.chains[chain], exclude_local_chains))
+        return chain_ids
+
+    def filter_chain_ids_by_name(
+        self, chain_name: str, exclude_local_chains: bool = True
+    ) -> list[ChainId]:
+        return self._get_network_chain_ids(self.chains[chain_name], exclude_local_chains)
+
+
+BEAMER_CONFIG = BeamerConfig(
+    chains={
+        "ethereum": {"mainnet": ChainId(1), "goerli": ChainId(5), "local": ChainId(1337)},
+        "optimism": {"mainnet": ChainId(10), "goerli": ChainId(420), "local": ChainId(901)},
+        "arbitrum": {
+            "mainnet": ChainId(42161),
+            "goerli": ChainId(421613),
+            "local": ChainId(412346),
+        },
+        "polygon_zkevm": {
+            "mainnet": ChainId(1101),
+            "goerli": ChainId(1442),
+            "local": ChainId(1001),
+        },
+        "base": {"mainnet": None, "goerli": ChainId(84531), "local": None},
+    },
+    type2_chains=("ethereum", "optimism"),
+    optimistic_chains=("optimism", "base"),
```

Comment:
would rename to `op_stack_chains` or even more explicit `bedrock_chains` if you check for that in `maybe_prove`